ext/mysql2/client.c

```diff
@@ -880,7 +880,7 @@ static VALUE rb_mysql_client_server_info(VALUE self) {
  *
  * Return the file descriptor number for this client.
  */
-static VALUE rb_mysql_client_socket(VALUE self) {
+static VALUE rb_mysql_client_socket(RB_MYSQL_UNUSED VALUE self) {
```

Comment:
I don't want to mark this as unused in all cases. It may make sense to move the ifdef outwards, and have two separate function definitions for windows and non-windows.